smac/epm/rfr_imputator.py

```diff
@@ -87,51 +88,64 @@ def impute(self, censored_X, censored_y, uncensored_X, uncensored_y):
             self.logger.debug("Iteration %d of %d" % (it, self.max_iter))
 
             # predict censored y values
             y_mean, y_var = self.model.predict(censored_X)
             y_var[y_var < self.var_threshold] = self.var_threshold
-            y_stdev = numpy.sqrt(y_var)[:,0]
-            y_mean = y_mean[:,0]
-
-            imputed_y = truncnorm.stats(a=(censored_y - y_mean) / y_stdev,
-                                    b=(self.cutoff * 10 - y_mean) / y_stdev,
-                                    loc=y_mean,
-                                    scale=y_stdev,
-                                    moments='m')
-            
-            imputed_y = numpy.array(imputed_y)
-
-            if sum(numpy.isfinite(imputed_y) == False) > 0:
-                # Replace all nans with threshold
+            y_stdev = np.sqrt(y_var)[:, 0]
+            y_mean = y_mean[:, 0]
+
+            # ignore the warnings of truncnorm.stats
+            # since we handle them appropriately
+            with warnings.catch_warnings():
+                warnings.filterwarnings(
+                    'ignore', r'invalid value encountered in (subtract|true_divide).*')
+                warnings.filterwarnings(
+                    'ignore', r'divide by zero encountered in (true_divide|log).*')
+                imputed_y = truncnorm.stats(a=(censored_y - y_mean) / y_stdev,
+                                            b=(self.threshold - y_mean) /
+                                            y_stdev,
+                                            loc=y_mean,
+                                            scale=y_stdev,
+                                            moments='m')
+
+            imputed_y = np.array(imputed_y)
+
+            nans = np.isfinite(imputed_y) == False
+            n_nans = sum(nans)
+            if n_nans > 0:
+                # Replace all nans with maximum of predicted perf and censored value
+                # this happens if the prediction is far smaller than the
+                # censored data point
                 self.logger.debug("Going to replace %d nan-value(s) with "
-                                  "threshold" %
-                                  sum(numpy.isfinite(imputed_y) == False))
-                imputed_y[numpy.isfinite(imputed_y) == False] = self.threshold
+                                  "max(captime, predicted mean)" % n_nans)
+                imputed_y[nans] = np.max(
+                    [censored_y[nans], y_mean[nans]], axis=0)
 
-            if it > 0:
+            if it > 1:
                 # Calc mean difference between imputed values this and last
                 # iteration, assume imputed values are always concatenated
                 # after uncensored values
 
-                change = numpy.mean(abs(imputed_y -
-                                        y[uncensored_y.shape[0]:]) /
-                                    y[uncensored_y.shape[0]:])
+                change = np.mean(abs(imputed_y -
```

Comment:
This works? I would have assumed that you need to use `np.abs` instead of `abs`.